stream_alert/rule_processor/payload.py

```diff
@@ -0,0 +1,328 @@
+            if line_num and line_num % 100 == 0:
+                avg_record_size = ((processed_size - 1) / line_num)
+                if avg_record_size:
+                    approx_record_count = self.s3_object_size / avg_record_size
+                    LOGGER.debug(
+                        'Processed %s records out of an approximate total of %s '
+                        '(average record size: %s bytes, total size: %s bytes)',
+                        line_num,
+                        approx_record_count,
+                        avg_record_size,
+                        self.s3_object_size)
+
+    def _download_object(self, region, bucket, key):
+        """Download an object from S3.
+
+        Verifies the S3 object is less than or equal to 128MB, and
+        stores into a temp file.  Lambda can only execute for a
+        maximum of 300 seconds, and the file to download
+        greatly impacts that time.
+
+        Args:
+            region [string]: AWS region to use for boto client instance.
+            bucket (string): S3 bucket to download object from.
+            key [string]: Key of s3 object.
+
+        Returns:
+            [string] The downloaded path of the S3 object.
+        """
+        size_kb = self.s3_object_size / 1024.0
+        size_mb = size_kb / 1024.0
+        if size_mb > 128:
+            raise S3ObjectSizeError('S3 object to download is above 128MB')
+
+        LOGGER.debug('/tmp directory contents:%s ', os.listdir('/tmp'))
+        LOGGER.debug(os.popen('df -h /tmp | tail -1').read().strip())
+
+        if size_mb:
+            display_size = '{}MB'.format(size_mb)
+        else:
+            display_size = '{}KB'.format(size_kb)
+
+        LOGGER.info('Starting download from S3: %s/%s [%s]', bucket, key, display_size)
+
+        suffix = key.replace('/', '-')
+        _, downloaded_s3_object = tempfile.mkstemp(suffix=suffix)
+        with open(downloaded_s3_object, 'wb') as data:
+            client = boto3.client('s3', region_name=region)
+            start_time = time.time()
+            client.download_fileobj(bucket, key, data)
+
+        total_time = time.time() - start_time
+        LOGGER.info('Completed download in %s seconds', round(total_time, 2))
+
+        return downloaded_s3_object
+
+    def _get_object(self):
+        """Given an S3 record, download and parse the data.
+
+        Returns:
+            [string] Path to the downloaded s3 object.
+        """
+        unquoted = lambda data: unquote(data).decode('utf8')
+        region = self.raw_record['awsRegion']
+        bucket = unquoted(self.raw_record['s3']['bucket']['name'])
+        key = unquoted(self.raw_record['s3']['object']['key'])
+        self.s3_object_size = int(self.raw_record['s3']['object']['size'])
+
+        LOGGER.debug('Pre-parsing record from S3. Bucket: %s, Key: %s, Size: %d',
+                     bucket, key, self.s3_object_size)
+
+        downloaded_s3_object = self._download_object(region, bucket, key)
```

Comment:
maybe just `return self._download_object(region, bucket, key)`, that var isn't used elsewhere